modules/wri_event_callout/src/Plugin/Field/FieldFormatter/EventCalloutBehaviorFormatter.php

```diff
@@ -0,0 +1,259 @@
+
+    $elements['view_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View ID'),
+      '#default_value' => $this->getSetting('view_id'),
+      '#description' => $this->t('Machine name of the View used to fetch the next upcoming event by topic.'),
+      '#required' => TRUE,
+    ];
+
+    $elements['display_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View display ID'),
+      '#default_value' => $this->getSetting('display_id'),
+      '#description' => $this->t('Display ID (for example "block_1") of the View.'),
+      '#required' => TRUE,
+    ];
+
+    $elements['event_date_field'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Event date field'),
+      '#default_value' => $this->getSetting('event_date_field'),
+      '#description' => $this->t('Machine name of the date or daterange field used to decide if the event is upcoming.'),
+      '#required' => TRUE,
+    ];
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary(): array {
+    $summary = parent::settingsSummary();
+
+    $summary[] = $this->t('View: @view (@display)', [
+      '@view' => $this->getSetting('view_id'),
+      '@display' => $this->getSetting('display_id'),
+    ]);
+
+    $summary[] = $this->t('Event date field: @field', [
+      '@field' => $this->getSetting('event_date_field'),
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode): array {
+    $elements = [];
+
+    $entity = $items->getEntity();
+    if (!$entity instanceof BlockContentInterface || $entity->bundle() !== 'upcoming_event') {
+      return $elements;
+    }
+
+    // Behavior value:
+    // - show_next_event
+    // - link_to_recording
+    // - hide_block
+    // Defaults to show_next_event (matches field default).
+    $behavior = $items->isEmpty() ? 'show_next_event' : (string) $items->first()->value;
+
+    /** @var \Drupal\node\NodeInterface|null $event */
+    $event = NULL;
+    if ($entity->hasField('field_event_to_feature') && !$entity->get('field_event_to_feature')->isEmpty()) {
+      $event = $entity->get('field_event_to_feature')->entity;
+    }
+
+    // Case 1: no explicit event chosen, always use "next upcoming by topic".
+    if (!$event instanceof NodeInterface) {
+      $build = $this->buildUpcomingByTopic($entity);
+      if (!empty($build)) {
+        $elements[0] = $build;
+      }
+      return $elements;
+    }
+
+    // Case 2: explicit event exists and is in the future, always show it.
+    if ($this->isEventUpcoming($event)) {
+      $elements[0] = $this->buildEventCallout($event);
+      return $elements;
+    }
+
+    // From here on, the event is over.
+    switch ($behavior) {
+      case 'link_to_recording':
+        // Show the chosen (past) event in Callout view mode.
+        $elements[0] = $this->buildEventCallout($event);
```

Comment:
There needs to be some logic here making sure the `field_body_contains_recording` value is true.